src/main/java/uk/gov/hmcts/ccd/domain/model/common/HttpError.java

```diff
@@ -23,29 +25,59 @@
     private List<String> callbackErrors;
     private List<String> callbackWarnings;
 
-    public HttpError(Exception exception, HttpServletRequest request) {
-        final ResponseStatus responseStatus = exception.getClass().getAnnotation(ResponseStatus.class);
+    private final transient CatalogueResponse catalogueResponse;
+
+
+    public HttpError(final Exception exception, final HttpServletRequest request) {
+
+        this(exception, request, null, null);
+    }
+
+    public HttpError(final ApiException exception, final HttpServletRequest request) {
 
+        this(exception, request, exception.getCatalogueResponse(),null);
+    }
+
+    public HttpError(
+        final Exception exception,
+        final HttpServletRequest request,
+        final CatalogueResponse catalogueResponse,
+        final Integer status) {
+
+        final ResponseStatus responseStatus = exception.getClass().getAnnotation(ResponseStatus.class);
         this.exception = exception.getClass().getName();
         this.timestamp = LocalDateTime.now(ZoneOffset.UTC);
-        this.status = getStatus(responseStatus);
-        this.error = getErrorReason(responseStatus);
+        this.status = getStatusValue(status, responseStatus);
+        this.error = getErrorReason(status, responseStatus);
         this.message = exception.getMessage();
         this.path = request.getRequestURI();
+        this.catalogueResponse = catalogueResponse;
     }
 
-    private Integer getStatus(ResponseStatus responseStatus) {
-        if (null != responseStatus) {
+    private Integer getStatusValue(final Integer status, final ResponseStatus responseStatus) {
+        // Option 1: use status as it may be an override value for responseStatus
+        if (status != null) {
+            return status;
+
+        // Option 2: use responseStatus
+        } else if (null != responseStatus) {
```

Comment:
I got what this method is doing now. But, if status codes annotated for Exceptions take precedence, couldn't it be a programming error to provide a status value for an Exception that already has an inconsistent annotated status?